lib/fluent/plugin/kafka_producer_ext.rb

```diff
@@ -10,16 +10,17 @@
 # for out_kafka_buffered
 module Kafka
   class Producer
-    def produce2(value, key: nil, topic:, partition: nil, partition_key: nil)
+    def produce_for_buffered(value, key: nil, topic:, partition: nil, partition_key: nil)
       create_time = Time.now
 
       message = PendingMessage.new(
```

Comment:
Should apply same changes to `TopicProducer#produce`?

Reply:
I did change there, but not quite sure if should use produce() or produce_for_buffered() there.
For now using produce_for_buffered, but if I'm mistaken, please do tell me and I'll correct it asap ;)

Reply:
I mean `out_kafka2` uses specific producer named `TopicProducer` and it also uses `PendingMessage` in its method: https://github.com/fluent/fluent-plugin-kafka/blob/91da31930601ad1a60f1d3bf9b32f3a509341ee3/lib/fluent/plugin/kafka_producer_ext.rb#L84

Reply:
Thanks for pointing it out ;)